app/components/views/TransactionsPage/HistoryTab/index.js

```diff
@@ -100,6 +100,17 @@ class History extends React.Component {
   }
 
   onChangeFilter(value) {
+    const { isChangingFilter } = this.state;
+    if (isChangingFilter) {
```

Comment:
Not quite there yet.

Notice in this file that `isChangingFilter` is not set to true anywhere - meaning that this code path is never exercised. Also, you're scheduling the `changeFilter` in two places.

Roughly what needs to happen is the following:

```javascript
OnChangeFilter(value) {
  const { isChangingFilterTimer } = this.state;
  if (isChangingFilterTimer) {
    clearTimeout(isChangingFilterTimer);
  }
  this.setState({ isChangingFilterTimer: setTimeout(() => this.changeFilter(value), 2000) });
}
````

Notice the following:

- `isChangingFilterTimer` is the handle to the most recently started timeout
- If no changes happen until the timeout (2000ms in this example), then the `changeFilter` is triggered and an update is requested
- If new changes happen, then the previous timeout is canceled and a new one started from zero

The point here is that, within the 2000ms interval many changes to the values can take place but only a single `changeFilter` must be triggered at the end of this.

You'll also need to clear the `isChangingFilterTimer` in the `changeFilter` (omitted here for brevity).